bsp/src/mill/bsp/MillBuildServer.scala

```diff
@@ -335,6 +341,15 @@ class MillBuildServer(
         targetIds = sourcesParams.getTargets.asScala.toSeq,
         agg = (items: Seq[SourcesItem]) => new SourcesResult(items.asJava)
       ) {
+        case (id, _: MillBuildTarget) if clientIsIntelliJ =>
+          T.task {
+            val sources = new SourcesItem(
+              id,
+              Seq(sourceItem(evaluator.rootModule.millSourcePath / ".bsp", false)).asJava
```

Comment:
This can be any subdirectory, but we chose the `.bsp` directory, which definitely exists and is set to ignore.

Reply:
Instead we could use some generated source directory under `out` which we define in `MillBuildTarget`. 

```scala
// in MillBuildTarget
/** Used in BSP IntelliJ, which can only work with directories */
def dummySources: Sources = T.sources(T.dest)
```

Reply:
Review applied.